src/query/server/server.go

```diff
@@ -357,6 +395,23 @@ func newStorages(
 	return fanoutStorage, cleanup
 }
 
+func remoteClient(
+	logger *zap.Logger,
+	cfg config.Configuration,
+) (storage.Storage, bool) {
+	if remotes := cfg.RPC.RemoteListenAddresses; len(remotes) > 0 {
+		client, err := tsdbRemote.NewGrpcClient(remotes)
+		if err != nil {
+			logger.Fatal("unable to start remote clients for addresses", zap.Any("error", err))
```

Comment:
nit: don't Fatalf in subfunctions, pass the error back up and fatal in `main` or the top-level `Run`